app/reducers/entityReducers/article.js

```diff
@@ -24,14 +24,14 @@ export default (state, action) => {
     }
     case UPDATE_ARTICLE_SUCCESS: {
       const article = action.payload.getIn(['entities', 'article']).first();
-      const articleUrl = state.getIn(['article']).findKey((el) => el.get('id') === action.meta.articleId);
 
-      return state.setIn(['article', articleUrl], article);
+      return state.update('article', (articles) =>
+                          articles.map((v) =>
+                                       (v.get('id') === action.meta.articleId ? article : v)));
```

Comment:
I'd better restructure our store in the following way:
- store articles by id
- for each URL, store associated article id

In that way, we can update the article in amortized O(1) and still update all tabs that show the article.